src/components/Tooltip/index.js

```diff
@@ -0,0 +1,202 @@
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import styles from './style.css';
+
+import Icon from '../Icon';
+
+export const directions = ['top', 'left', 'right', 'bottom'];
+
+const Tooltip = ({
+  children,
+  closeable,
+  closeOnEscape,
+  content: tooltipContent,
+  direction,
+  isOpen,
+  showOnHover,
+  tooltipProps: tooltipOptsProp,
+  tooltipWidth,
+  ...rest
+}) => {
+  const [tooltipId] = useState(
+    `tooltip-${Date.now() * Math.floor(100 * Math.random())}`
+  );
+  const tooltipRef = useRef(null);
+
+  const [hovered, setHovered] = useState(false);
+  const [tooltipHeight, setTooltipHeight] = useState(0);
+
+  const closeFunc = useCallback(closeable || function() {}, []);
+
+  const handleSetHover = value => () => {
+    if (showOnHover) {
+      setHovered(value);
+    }
+  };
+
+  const handleEscape = e => {
+    if (closeable && e.key === 'Escape') {
+      closeable();
+    }
+  };
+
+  useEffect(() => {
+    if (closeOnEscape) {
+      document.addEventListener('keydown', handleEscape);
+
+      return () => {
+        document.removeEventListener('keydown', handleEscape);
+      };
+    }
+
+    return () => {};
+  }, []);
+
+  useEffect(() => {
```

Comment:
should we use `useLayoutEffect` instead, so we set the height before the browser's paint step? I've found that by using `useLayoutEffect`, for these cases where we need the content's width/height, we avoid flickering on the screen (we might need it if the tooltip is opened when rendering)

Reply:
Oh, that's actually something I hadn't considered. `useLayoutEffect` is still unknown to me, so time to dive in!